fabtests/multinode/src/core.c

```diff
@@ -281,21 +281,30 @@ static void pm_job_free_res()
 		free(pm_job.names);
 
 	if (pm_job.fi_addrs)
-	free(pm_job.fi_addrs);
+		free(pm_job.fi_addrs);
 }
 
 int multinode_run_tests(int argc, char **argv)
 {
 	int ret = FI_SUCCESS;
+	int i;
 
 	ret = multinode_setup_fabric(argc, argv);
 	if (ret)
 		return ret;
 
-	pattern = &full_mesh_ops;
-
-	ret = multinode_run_test();
-
+	for (i = 0; i < NUM_TESTS; i++) {
+		printf("starting %s... ", patterns[i].name);
+		pattern = &patterns[i];
+		ret = multinode_run_test();
+		if (ret) {
+			printf("failed\n");
+			return ret;
```

Comment:
This skips the cleanup code after the for loop.  I would remove the return and add '&& !ret' as part of the for loop check